app/src/main/java/com/hedvig/app/feature/offer/OfferViewModel.kt

```diff
@@ -1,22 +1,26 @@
 package com.hedvig.app.feature.offer
 
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import com.apollographql.apollo.api.Response
 import com.hedvig.android.owldroid.fragment.SignStatusFragment
 import com.hedvig.android.owldroid.graphql.OfferQuery
 import com.hedvig.android.owldroid.graphql.RedeemReferralCodeMutation
 import com.hedvig.android.owldroid.graphql.SignOfferMutation
+import com.hedvig.app.feature.offer.ui.OfferModel
 import e
 import kotlinx.coroutines.flow.catch
-import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.firstOrNull
 import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.flow.onEach
 import kotlinx.coroutines.launch
 import java.time.LocalDate
 
 abstract class OfferViewModel : ViewModel() {
-    abstract val data: MutableLiveData<OfferQuery.Data>
+    abstract val viewState: LiveData<ViewState>
```

Comment:
Let's make this non-abstract - the base class can contain both the producer and consumer-handles of the `LiveData`, as such:

```kotlin
protected val _viewState = MutableLiveData<ViewState>()
val viewState: LiveData<ViewState> = _viewState
```